engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -887,4 +891,13 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles): Future[WorkflowDescriptor] = {
```

Comment:
Well, it `validates` as a side effect now (hence we changed the name that day), mostly it creates a WorkflowDescriptor from sources.